pontoon/base/migrations/0116_fix_ambiguous_entity_order.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.8 on 2018-02-09 14:02
+from __future__ import unicode_literals
+
+from collections import Counter
+from itertools import dropwhile
+
+from django.db import migrations
+
+
+def _find_resources_with_dupes(apps):
+    Entity = apps.get_model('base', 'Entity')
+
+    entities = Counter(Entity.objects.values_list('resource', 'order').iterator())
+    for k, c in dropwhile(lambda v: v[1] > 1, entities.most_common()):
+        del entities[k]
+    return [x[0] for x in entities.keys()]
+
+
+def fix_ambiguous_entity_order(apps, schema_editor):
+    Entity = apps.get_model('base', 'Entity')
+    entities = Entity.objects.filter(
+        resource_id__in=_find_resources_with_dupes(apps))
+    resource = None
+    order = 0
+    fixed_entities = {}
+    for _resource, pk in entities.order_by('resource', 'order').values_list('resource', 'pk').iterator():
+        if resource is None:
+            resource = _resource
+        elif resource != _resource:
+            # update on the boundary
+            order = 0
+            resource = _resource
+        else:
+            order += 1
+        fixed_entities[order] = fixed_entities.get(order, [])
+        fixed_entities[order].append(pk)
+
+    for order, pks in fixed_entities.items():
+        Entity.objects.filter(pk__in=pks).update(order=order)
```

Comment:
In theory we should reset the order based on the order in the actual file.

Anyhow, since the inconsistencies are [not that important,](https://github.com/mozilla/pontoon/pull/846#issuecomment-365400510) we should be good.

It would be nice to add a comment explaining how the algorithm actually works.

Reply:
ill add some comments